x-pack/plugins/rule_registry/server/routes/get_alerts_group_aggregations.test.ts

```diff
@@ -0,0 +1,261 @@
+          },
+        ],
+      },
+      unitsCount: {
+        value: 185,
+      },
+    },
+  } as unknown as ReturnType<AlertsClient['getGroupAggregations']>);
+
+describe('getAlertsGroupAggregations', () => {
+  let server: ReturnType<typeof serverMock.create>;
+  let { clients, context } = requestContextMock.createTools();
+
+  beforeEach(async () => {
+    server = serverMock.create();
+    ({ clients, context } = requestContextMock.createTools());
+
+    clients.rac.getGroupAggregations.mockResolvedValue(getMockAggregations());
+
+    getAlertsGroupAggregations(server.router);
+  });
+
+  test('returns 200 when the parameters are correct', async () => {
+    const response = await server.inject(getAlertsGroupAggregationsRequest(), context);
+
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(getMockAggregations());
+  });
+
+  describe('request validation', () => {
+    test('rejects with missing params', async () => {
+      await expect(
+        server.inject(
+          requestMock.create({
+            method: 'post',
+            path: `${BASE_RAC_ALERTS_API_PATH}/_group_aggregations`,
+            body: {
+              groupByField: 'kibana.alert.rule.name',
+            },
+          }),
+          context
+        )
+      ).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"Request was rejected with message: 'Invalid value \\"undefined\\" supplied to \\"featureIds\\"'"`
+      );
+    });
+
+    test('rejects with invalid params', async () => {
+      await expect(
+        server.inject(
+          requestMock.create({
+            method: 'post',
+            path: `${BASE_RAC_ALERTS_API_PATH}/_group_aggregations`,
+            body: {
+              featureIds: ['apm', 'infrastructure', 'logs', 'observability', 'slo', 'uptime'],
+              groupByField: 'kibana.alert.rule.name',
+              aggregations: {
+                scriptedAggregation: {
+                  script: '100',
+                },
+              },
+            },
+          }),
+          context
+        )
+      ).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"Request was rejected with message: 'invalid keys \\"script\\"'"`
+      );
+    });
+
+    test('rejects unknown query params', async () => {
+      await expect(
+        server.inject(
+          requestMock.create({
+            method: 'post',
+            path: `${BASE_RAC_ALERTS_API_PATH}/_group_aggregations`,
+            body: {
+              featureIds: ['apm', 'infrastructure', 'logs', 'observability', 'slo', 'uptime'],
+              groupByField: 'kibana.alert.rule.name',
+              aggregations: {},
+              runtimeMappings: {},
+            },
+          }),
+          context
+        )
+      ).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"Request was rejected with message: 'invalid keys \\"runtimeMappings,{}\\"'"`
+      );
+    });
+  });
+
+  test('returns error status if rac client find fails', async () => {
```

Comment:
Could you also add a test where `getGroupAggregations` resolves but `alerts == null`? If I understand correctly the message should be different right? Satus 404 and message `alerts not found`?

Reply:
As discussed on Slack that case is unreachable without an exception being thrown, I've removed the check altogether